doc/contribute/committer_expectations.rst

```diff
@@ -0,0 +1,272 @@
+viable implementation.
+
+Changes which require an RFC proposal include:
+
+- Submitting a new feature.
+- Submitting a new API.
+- :ref:`treewide-changes`.
+- Other large changes that can benefit from the RFC proposal process.
+  Maintainers have the discretion to request you create an RFC for your change.
+
+PR Requirements
+***************
+
+- Each commit in the PR must provide a commit message following the
+  :ref:`commit_message_guidelines`.
+
+- All files in the PR must comply with :ref:`Licensing
+  Requirements<licensing_requirements>`.
+
+- Follow the Zephyr :ref:`coding_style` and :ref:`coding_guidelines`.
+
+- PRs must pass all CI checks. This is a requirement to merge the PR. You may
+  mark a PR as draft and explicitly request reviewers to provide early feedback,
+  even with failing CI checks.
+
+- When breaking a PR into multiple commits, each commit must build cleanly. The
+  CI system does not enforce this policy, so it is the PR author's
+  responsibility to verify.
+
+- When major new functionality is added, tests for the new functionality shall
+  be added to the automated test suite. All API functions should have test cases
+  and there should be tests for the behavior contracts of the API. Maintainers
+  and reviewers have the discretion to determine if the provided tests are
+  sufficient. The examples below demonstrate best practices on how to test APIs
+  effectively.
+
+    - `Kernel timer tests`_ provide better than 85% `test coverage`_ for the
+      kernel timer.
+    - TODO: Provide a driver level API test example.
+
+- Incompatible changes to APIs must also update the release notes for the
+  next release detailing the change.  APIs marked as experimental are excluded
+  from this requirement. [#api-updates]_
+
+- Changes to APIs must increment the API version number according to the API
+  version rules. [#api-version]_
+
+- PRs must also satisfy all :ref:`merge_criteria` before a member of the release
+  engineering team merges the PR into the zephyr tree.
+
+Maintainers may request you break up a PR into smaller PRs and may request you
+create an :ref:`RFC proposal <rfcs>`.
+
+.. _Kernel timer tests: https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/kernel/timer/timer_behavior
+
+.. _test coverage: https://app.codecov.io/gh/zephyrproject-rtos/zephyr/blob/main/kernel/timer.c
+
+Workflow Suggestions That Help Reviewers
+========================================
+
+- Don't resolve comments until the question has been answered or the request
+  completed in an update to the PR.
+
+- Respond to comments from the "Files changed" view. This allows you to respond
+  to multiple comments and publish the responses in bulk. This reduces the
+  number of emails sent to reviewers.
+
+- Try to minimize rebases in the middle of a review. If a rebase is required,
+  push this as a separate update with no functional changes to the PR. When you
```

Comment:
Non-blocking: I would say, "no other changes since the last push of the PR." Many changes of the type that people ask for in code reviews will not be functional.